### ISSUE TITLE
0.12 beta 2: "Internal validation of the provider failed"

### ISSUE THREAD

**Issue by maur1th** [terraform-providers/terraform-provider-vault#410]
### Terraform Version
```
Terraform v0.12.0-beta2
+ provider.vault v1.6.0-dev20190216h01-dev
```

### Affected Resource(s)
The whole Vault provider apparently.

### Terraform Configuration Files
```hcl
provider "vault" {
  address         = "endpoint_url"
  skip_tls_verify = false
}

resource "vault_generic_secret" "example" {
  path = "secret/foo"

  data_json = jsonencode({
    "foo" : "bar"
    "hello" : "world"
  })
}
```

### Debug Output
https://gist.github.com/maur1th/afcc5e8eefcbc56021d23dbc28daed87

### Expected Behavior
Should have created a secret in Vault.

### Actual Behavior
```
Error: Internal validation of the provider failed! This is always a bug
with the provider itself, and not a user issue. Please report
this bug:

4 errors occurred:
	* resource vault_identity_group_alias: id is a reserved field name
	* resource vault_identity_group: id is a reserved field name
	* resource vault_identity_entity: id is a reserved field name
	* resource vault_identity_entity_alias: id is a reserved field name
```

### Steps to Reproduce
`terraform apply`

**Comment by maur1th** [comment]
Tried with rc1 as well, same results.

**Comment by EliaSaSe** [comment]
Hi  @maur1th 

For testing purposes, I'm trying to upgrade to terraform 0.12rc1.

When I'm using ``Terraform v0.12.0-rc1`` with the vault provider (build from master today), I'm getting the error ``Error: Failed to instantiate provider "vault" to obtain schema: Incompatible API version with plugin. Plugin version: 4, Client versions: [5]``. So this provider is not ready for terraform 0.12? 

Why do you not have this api version error with ``provider.vault v1.6.0-dev20190216h01-dev``?
Where did you get this build?

Best regards
Elia

**Comment by maur1th** [comment]
Hi @EliaSaSe, I got it from the [development snapshots S3 bucket](http://terraform-0.12.0-dev-snapshots.s3-website-us-west-2.amazonaws.com/) mentioned in the [Terraform 0.12 beta 2 announcement](https://www.hashicorp.com/blog/announcing-terraform-0-1-2-beta-2).

**Comment by EliaSaSe** [comment]
Hi @maur1th, thank you for the urls to this build. I can reproduce this errors to. It seems it's needs some breaking changes in this provider, when it's using fieldnames that are reserved in terraform 0.12.

**Comment by paultyng** [comment]
Fixes for this should be in #446

**Comment by paultyng** [comment]
#446 is merged and should be going out shortly.
